client/hotrod/src/main/java/org/infinispan/hotrod/near/NearCacheService.java

```diff
@@ -111,19 +111,25 @@ public static <K, V> NearCacheService<K, V> create(
    }
 
    @Override
-   public void put(CacheEntry<K, V> entry) {
-       cache.put(entry);
+   public boolean putIfAbsent(K key, CacheEntry<K, V> entry) {
+      boolean inserted = cache.putIfAbsent(key, entry);
 
       if (log.isTraceEnabled())
-         log.tracef("Put %s in near cache (listenerId=%s)", entry, Util.printArray(listenerId));
+         log.tracef("Conditionally put %s if absent in near cache (listenerId=%s)", entry, Util.printArray(listenerId));
+      return inserted;
    }
 
    @Override
-   public void putIfAbsent(CacheEntry<K, V> entry) {
-      cache.putIfAbsent(entry);
-
-      if (log.isTraceEnabled())
-         log.tracef("Conditionally put %s if absent in near cache (listenerId=%s)", entry, Util.printArray(listenerId));
+   public boolean replace(K key, CacheEntry<K, V> prevValue, CacheEntry<K, V> newValue) {
+      boolean replaced = cache.replace(key, prevValue, newValue);
+      if (replaced) {
+         log.tracef("Replaced key=%s and value=%s with new value=%s in near cache (listenerId=%s)",
```

Comment:
probably worth merging into a single log message with `replaced` includes.
nitpicking: checking `log.isTraceEnabled()` missing. Same comment for the "other NearCacheService"